### PR TITLE
Add proper subcommand system, Cleanup CommandManager code

### PR DESCRIPTION
**Pull request by dlsf**
I also shortened the code that registers all commands. I know that @PeachesMLG didn't want this last time but I managed to find a solution without using a third-party library that is in my opinion way better than the manual registration. I can revert that part if it's still not wanted.

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/commands/BankCommand.java [hunk 1]
```diff
@@ -24,18 +24,20 @@
  */
 public class BankCommand extends Command {
 
-    public BankGive bankGive;
-    public BankSet bankSet;
-    public BankRemove bankRemove;
+    public GiveCommand bankGive;
```
[the remark this answers is not in the document]
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/commands/BankCommand.java:27]: These variable names are in the config, renaming them here will mess up existing configs and confuse people.

**Comment by dlsf** [comment]
I obviously tested this PR, except child commands having other children.

File: src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java [hunk 2]
```diff
@@ -233,16 +196,40 @@ public boolean onCommand(@NotNull CommandSender commandSender, org.bukkit.comman
...
+            || command.permission.equalsIgnoreCase("")
+            || command.permission.equalsIgnoreCase("iridiumskyblock.");
+    }
+
+    private Command findExecutingCommand(Command baseCommand, String[] args) {
+        Command executingCommand = baseCommand;
+        int currentArgument = 1;
+
+        // Check for each argument if it's a child of the previous command
+        while (args.length > currentArgument) {
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java:224]: Cant this just be a for loop? or am I being dum?

Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java]: Well, yeah

**Pull request merged**